testing/build.gradle

```diff
@@ -0,0 +1,50 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+apply plugin: 'kotlin-kapt'
+
+android {
```

Comment:
I don't see why this needs to be an android module?

Reply:
The classes included in this module depend on test-only libraries like Robolectric. We can't actually include those in the production builds of the app, and any source files included in the other modules are automatically included unless they're under the test folder. Since these utilities aren't actually tests, they shouldn't go under the test folder. It seems like having a dedicated, test-only module is the only way to solve this problem. Do you have another approach that might work?

Reply:
I'm curious, have we already specified in this PR (if so, can you tell me where?) that for release builds we shouldn't include this module?  Or is this something we would like to do in the future/something gradle does by default? I am wondering how to go about that in gradle builds

Reply:
It's an implicit thing right now--the project isn't included for any non-test dependency lists for the other modules so it won't be included. With modules we have control over what dependencies are included, but not over what sources are included which is why we needed to make this split.

I *think* it might be possible to segment sources within a single module, but I'm expecting that to cause other issues since it's a rather non-typical thing to do with Gradle. This seemed like a more Gradle-idiomatic way to ensure certain source files aren't included.

Reply:
@aggarwalpulkit596 since you didn't follow up, I'm taking that to mean you're okay with the test-only module. Please follow up with me if you're not. Since this is blocking a lot of downstream work, I want to get this submitted.